force-app/main/default/classes/ODRIntegration.cls

```diff
@@ -387,7 +387,30 @@ public with sharing class ODRIntegration {
         system.debug('The response is :'+ sar.saRecords);
         return sar;
     }
-
+    
+    @AuraEnabled(cacheable = true)
+    public static PrescriptionHistoryResponse fetchPrescriptionHistoryWithSearchKey(Id recordId, String page, String totalCount, List<Integer> dinList, String searchKey, String displayCount) {
+        if(String.isBlank(searchKey)){
+            return fetchPrescriptionHistory(recordId, page, displayCount, dinList);
+        }
+        
+        PrescriptionHistoryResponse resObj = fetchPrescriptionHistory(recordId, page, totalCount, dinList);
+        Integer returnRecordCount = 0;
+        Integer displayCountInt = !String.isBlank(displayCount) ? Integer.valueOf(displayCount) : 0;
+        List<PrescriptionHistoryResponse.MedRecords> medRecordsList = new List<PrescriptionHistoryResponse.MedRecords>();
+        for(PrescriptionHistoryResponse.MedRecords medObj : resObj.medHistory.medRecords){
+            if(returnRecordCount < displayCountInt){
```

Comment:
Since the count are declared outside, why are we checking the condition within for loop? Do we have dependency on the iterated record to check the event every time? What if resObj.medHistory.medRecords has 10k records, will it check the count 10k times unless it is required which i don't think so?

Reply:
We are checking the conditions in for loop because we should retrieve only no. of records that we are showing in current page. For example we are showing only 10 records in a page, and there are more than 10 records with the given search key. So list will have only 10 records with the given search key.

Reply:
Even though the response has given 10k records. the for loop will iterate only for number of records that we need to show on the UI. if suppose 10k records are there in response and count of records we are showing on the UI is 10. Then for loop will iterate until we get 10 records with given search key. Once we get that, we are breaking the for loop in else condition... we have tested with 700 records  and it was passing there.